src/tests.rs

```diff
@@ -0,0 +1,117 @@
+//! Common tests and testing utilities
+extern crate std;
+
+use crate::Error;
+use std::{mem::MaybeUninit, sync::mpsc, thread, vec, vec::Vec};
+
+#[cfg(feature = "test-in-browser")]
+wasm_bindgen_test::wasm_bindgen_test_configure!(run_in_browser);
+
+fn num_diff_bits(s1: &[u8], s2: &[u8]) -> usize {
+    assert_eq!(s1.len(), s2.len());
+    s1.iter()
+        .zip(s2.iter())
+        .map(|(a, b)| (a ^ b).count_ones() as usize)
+        .sum()
+}
+
+// A function which fills a buffer with random bytes.
+type FillFn<B> = fn(&mut [B]) -> Result<(), Error>;
```

Comment:
I think this should be type `FillFn<B, R> = fn(&mut [B]) -> Result<R, Error>` so that all of `getrandom::getrandom_uninit`, `getrandom::getrandom`, and the internal `getrandom_impl` functions can all be used with it.

Reply:
Unfortunately, this doesn't work. Due to lifetime-related problems, there's no way to have a type alias which refers to the type of all 3 of those functions. Specifically, there's no `R` for which:
```rust
getrandom_uninit: FillFn<B,R>
```
as the full type is:
```rust
getrandom_uninit: for<'a> fn(&'a mut [MaybeUninit<u8>]) -> Result<&'a mut [u8], Error>
```

Instead, I changed things to have `FillFn` be a trait (implemented by the relevant `Fn` objects) which defines a `make_vec` method. The only downside is that it requires a dummy generic parameter to avoid conflicting implementations. Let me know what you think.